### PR TITLE
Add explicit info about PR review process & time

### LINKED ISSUE #224: More explicit merge guidelines in CONTRIBUTING?
I'd like to have more explicit metrics around when something is ready for merge. Currently, I worry both that (a) PR's aren't getting seen by the people who might want to review them before merge and (b) PR's just sit there even after an "approve" review or two because it's not clear who all should review them. It feels fairly no-win and concentrates the power of repo maintenance into people who already feel a lot of ownership over a given project.

In our [Contributing guidelines](https://github.com/edgi-govdata-archiving/overview/blob/master/CONTRIBUTING.md), we say:

> Allow others sufficient time for review and comments before merging. We make use of GitHub's review feature to comment in-line on PRs when possible. There may be some fixes or adjustments you'll have to make based on feedback.

But what is "sufficient time"? How long should we leave approved (or un-commented-on) reviews open?

### PR DESCRIPTION
**Pull request by Mr0grog**
We have a bunch of rules for PR reviewing/merging/timeframes we’ve talked through at various web monitoring meetings, but they’re otherwise just held in people’s (or maybe just my?) head. I’ve attempted to write them down more explicitly here. I’ve changed the format and organization from what I originally write in #224 so that it fits the structure we already had better, but let me know if that’s made it less clear.

Also totally open to a review of this process! It’s what we generally use in web monitoring, but we haven’t talked through it in a long time and never really wrote it down. This is as good an opportunity as any to poke holes in it. 😅

Fixes #224.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (patcon) [inline CONTRIBUTING.md]:
```suggestion
    - If you have commit rights, allow up to 3 days for review, after which you can merge it yourself (even without review).
    - If you have a urgent *hotfix* ([see below](#hotfixes)), please bug someone on Slack for rapid review, but you may merge without if nobody can.
    - If you have a drastic change (e.g., one that might break everyone’s development environment), consider allowing extra time beyond 3 days for others to review. There aren’t hard rules here—being given commit rights is a sign of trust that you’ll make reasonable decisions. When in doubt, opt for more time & feedback rather than less.
```

Went a little [_Strunk & White_](https://en.wikipedia.org/wiki/The_Elements_of_Style) on this :) and used bullets to pull scenarios out of paragraph.

Comment (Mr0grog) [reply CONTRIBUTING.md]: Hmmm, the second two points only apply in the context of having commit rights, though. I think we need a top bullet point (“- If you have commit irghts:”) to collect all these together. That’s why they were originally in one paragraph (totally appreciate breaking out the bullets, though).

Comment (patcon) [reply CONTRIBUTING.md]: ok, sounds right! mainly thought I could contribute as ruthless word cutter, so modify as appropriate -- or discard if you feel it loses a certain _"je ne sais quoi"_ (that's spanish for "hey, i liked those words") 😜

Comment (Mr0grog) [reply CONTRIBUTING.md]: How does 570b2de0d98943c2127fa3a95bf2aefcb1ffc6c4 read to you? I split up the first point to head up the bullets and to try and clarify that you shouldn’t just merge after 3 days if you’re still in the middle of a back-and-forth discussion. (Hopefully that’s obvious anyway, but a different reading jumped out at me when you framed it as a single bullet.)

Comment (Mr0grog) [reply CONTRIBUTING.md]: > mainly thought I could contribute as ruthless word cutter

That’s almost always super helpful, including here! I should have said “thanks, that really helps make this clearer and easier to read” first. :D

File: CONTRIBUTING.md
Comment (lightandluck) [inline CONTRIBUTING.md]: I've actually never "squash and merged" before. 😳 
> Choose what's correct for the situation

I'm not sure how to choose. Any resource we could link to that would that would help someone like me understand the differences between the choices?

Comment (Mr0grog) [reply CONTRIBUTING.md]: Not that I know of (each of these three are relatively standard git operations), but:

- “Merge” merges two histories, literally. This keeps a followable set of revisions involved in the PR fully intact once everything is merged onto the master branch, but can also make sorting through what-happened-when or what’s-related-to-what tough.

- “Rebase” takes all the current commits and replays them on top of the current commits on the `master` branch, then merges *that*. You get the fully intact history, but all the commits for a PR will come in sequence, instead of being interleaved with other PRs based on the time they were committed. You don’t get a single commit that represents the merging of the PR, though (because nothing is needed to “tie together” the two branches of code that are happening, since everything is now in order), which can make it hard to see when an actual PR landed or what it was about as a whole.

- “Squash & Merge” smashes all the commits in the PR together into a single commit, and then rebases *that* on top of the `master` branch. You get a nice single commit that represents everything that happened in PR, though you lose the full history of each commit. You also have an opportunity to revise the commit message when merging so it can be more representative of the complete work.

The actual commit history on a PR is often a little messy, so there’s usually more value in squashing it than in keeping it with a straight merge or rebase. It’s also often helpful to be able to see a single, clear commit for each PR in the history. On the other hand, if someone has carefully curated their commits (I used to spend a lot of effort on this), it may be useful to merge so you can see the work of a PR broken down in the `master` branch’s history.

I use “rebase” when merging single-commit PRs from Dependabot and pretty much in no other situation ever. I don’t generally need a reference back to the PR involved (since there’s rarely ever anything useful there) and since there’s only one commit, it’s clearer to just rebase instead of merging, which will sometimes result in two commits (the original commit and the “merge” commit that ties the two branches of code together). In almost all other situations I want a merge commit there to provide a clear point in the history that shows when the PR landed and the master branch and the PR code got brought together (or just one commit by squashing, which obviates the whole issue of two separate lines of history being joined together).

Comment (Mr0grog) [reply CONTRIBUTING.md]: (Actually, only two — “merge” and “rebase” — are standard git things. “Squash & merge” is the same as doing a special kind of rebase [`git rebase -i master`] followed by fast-forward merging onto master [`git checkout master && git merge --ff-only <your_PR_branch>`].)

Comment (Mr0grog) [reply CONTRIBUTING.md]: These pieces go into more depth:

- https://www.atlassian.com/git/tutorials/merging-vs-rebasing
- https://www.atlassian.com/git/articles/git-team-workflows-merge-or-rebase
- https://devblogs.microsoft.com/devops/squash-a-whole-new-way-to-merge-pull-requests/

Comment (Mr0grog) [reply CONTRIBUTING.md]: But mostly, if you don’t know how or whether to choose, just use “squash & merge.” It’s not worth thinking particularly hard about 99.9% of the time.

Comment (lightandluck) [reply CONTRIBUTING.md]: Thanks for the explication, that's really useful!

I guess I usually just tried to rebase using interactive mode locally to clean up my commit history before submitting a PR (not always consistent about this). In that case, I should just merge?

To clarify, I never used "squash and merge" (or any of the other options) _when merging a PR in Github._ Good to know when to use the different options when merging a PR. I think I'll have to play around in a test repo to really understand the differences.

Comment (lightandluck) [reply CONTRIBUTING.md]: But, yea, understood about generally using "squash and merge" for messier PRs.

I'll take a look at the commit history after this is merged too, to get a feel for things. Thanks for the info!

**Review by lightandluck — APPROVED** [review]
LGTM!

**Pull request merged**
